bundles/org.openhab.core.config.discovery/src/main/java/org/openhab/core/config/discovery/internal/DiscoveryResultImpl.java

```diff
@@ -48,44 +49,20 @@
     /**
      * Creates a new instance of this class with the specified parameters.
      *
-     * @param thingUID
-     *            the Thing UID to be set (must not be null). If a {@code Thing} disappears and is discovered again, the
-     *            same {@code Thing} ID
-     *            must be created. A typical {@code Thing} ID could be the
+     * @param thingTypeUID the {@link ThingTypeUID}
+     * @param thingUID the {@link ThingUID} to be set (must not be null). If a {@code Thing} disappears and is
```

Comment:
If you annotate the class with `@NonNullByDefault`  and methods with `@Nullable` (where applicable) you can also cleanup all the JavaDoc texts about parameters being (not) null. :wink:

Reply:
Done.